### PR TITLE
Add Azure Database offerings

### PR DESCRIPTION
**Pull request by teszes**
Adds Azure SQL and NoSQL offernings, namely:
- CosmosDB accounts
- SQL CosmosDB tables
- Cassandra CosmosDB tables
- Gremlin CosmosDB tables
- MongoDB CosmosDB tables
- Azure SQL databases
- MSSQL databases
- MySQL databases
- MariaDB databases
- PostgreSQL databases

### CODE REVIEW

**Comment by ianmiell** [comment]
LGTM
Will merge and resolve conflicts by hand.

File: azurerm/azurerm_mariadb_server/simple/main.tf [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+resource "azurerm_mariadb_server" "changeme_simple_mariadb_server" {
+  name                = "changeme-simple-mariadb-database-name"
+  resource_group_name = azurerm_resource_group.changeme_simple_mariadb_server_resource_group.name
+  location            = azurerm_resource_group.changeme_simple_mariadb_server_resource_group.location
+
+  storage_mb = "5120"
+  sku_name   = "B_Gen5_1"
+  version    = "10.3"
+
+  administrator_login          = "changeme-admin-username"
```
Comment (ttarczynski) [inline azurerm/azurerm_mariadb_server/simple/main.tf:45]: This has caused an apply error. See: https://github.com/ContainerSolutions/terraform-examples/runs/3220364029?check_suite_focus=true

```
/home/runner/work/terraform-examples/terraform-examples/bin/run_azurerm_examples.sh checking folder: azurerm/azurerm_mariadb_server/simple ... azurerm/azurerm_mariadb_server/simple
./run.sh/home/runner/work/terraform-examples/terraform-examples/azurerm/azurerm_mariadb_server/simple

...

azurerm_mariadb_server.changeme_simple_mariadb_server: Creating...
╷
│ Error: creating MariaDB Server "changeme-simple-mariadb-database-name" (Resource Group "changeme-simple-mariadb-database-resource-group"): mariadb.ServersClient#Create: Failure sending request: StatusCode=0 -- Original Error: Code="TokenTooLong" Message="The login that starts with 'changeme-admin-u' is too long. Maximum length is 16."
│ 
│   with azurerm_mariadb_server.changeme_simple_mariadb_server,
│   on main.tf line 36, in resource "azurerm_mariadb_server" "changeme_simple_mariadb_server":
│   36: resource "azurerm_mariadb_server" "changeme_simple_mariadb_server" ***
│ 
╵
/home/runner/work/terraform-examples/terraform-examples/azurerm/azurerm_mariadb_server/simple
```

**Review by ttarczynski — COMMENTED** [review]
`terraform apply` doesn't work for some examples.
Please make sure that `./run.sh` works successfully for all the new examples here.

**Comment by teszes** [comment]
Rebased from most recent main branch with added fix for Cassandra

**Comment by ttarczynski** [comment]
Another error when running `terraform apply`:
```
azurerm_postgresql_server.changeme_simple_postgresql_server: Creating...
╷
│ Error: creating PostgreSQL Server "changeme-simple-postgresql-database-name" (Resource Group "changeme-simple-postgresql-database-resource-group"): postgresql.ServersClient#Create: Failure sending request: StatusCode=0 -- Original Error: Code="InvalidUserName" Message="'changeme-admin-u' is not a valid name because it contains invalid characters."
│ 
│   with azurerm_postgresql_server.changeme_simple_postgresql_server,
│   on main.tf line 36, in resource "azurerm_postgresql_server" "changeme_simple_postgresql_server":
│   36: resource "azurerm_postgresql_server" "changeme_simple_postgresql_server" ***
│ 
╵
/home/runner/work/terraform-examples/terraform-examples/azurerm/azurerm_postgresql_server/simple
```

I'll add a fix for it.

**Review by ttarczynski — APPROVED** [review]
LGTM
All CI tests passed on integration.

**Pull request merged**